extension/src/editors/cms-editor/webview-communication.ts

```diff
@@ -0,0 +1,44 @@
+import type { CmsActionArgs } from '@axonivy/cms-editor-protocol';
+import { DisposableCollection } from '@eclipse-glsp/vscode-integration';
+import * as vscode from 'vscode';
+import { Messenger } from 'vscode-messenger';
+import { MessageParticipant, NotificationType } from 'vscode-messenger-common';
+import { IvyBrowserViewProvider } from '../../browser/ivy-browser-view-provider';
+import { InitializeConnectionRequest, isAction, WebviewReadyNotification } from '../notification-helper';
+import { WebSocketForwarder } from '../websocket-forwarder';
+
+const ConfigWebSocketMessage: NotificationType<unknown> = { method: 'cmsWebSocketMessage' };
+
+export const setupCommunication = (
+  websocketUrl: URL,
+  messenger: Messenger,
+  webviewPanel: vscode.WebviewPanel,
+  document?: vscode.TextDocument
+) => {
+  const messageParticipant = messenger.registerWebviewPanel(webviewPanel);
+  const toDispose = new DisposableCollection(
+    new CmsEditorWebSocketForwarder(websocketUrl, messenger, messageParticipant),
+    messenger.onNotification(
+      WebviewReadyNotification,
+      () =>
+        messenger.sendNotification(InitializeConnectionRequest, messageParticipant, {
+          file: document?.fileName ?? vscode.workspace.workspaceFolders?.[0].uri.path
```

Comment:
NB: vscode.workspace.workspaceFolders?.[0].uri.path does not necessary have to be a project
but ok for now I think.
possible improvement: add project selection dialog.